src/coreclr/vm/jitinterface.cpp

```diff
@@ -1453,6 +1453,41 @@ static CorInfoHelpFunc getInstanceFieldHelper(FieldDesc * pField, CORINFO_ACCESS
     return (CorInfoHelpFunc)helper;
 }
 
+static OBJECTREF getFrozenBoxedStatic(FieldDesc* field)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_COOPERATIVE;
+    } CONTRACTL_END;
+
+    // These fields never hold frozen boxed statics
+    if (!field->IsStatic() || !field->IsByValue() || field->IsSpecialStatic())
+    {
+        return NULL;
+    }
+
+    TypeHandle typeHandle = field->GetFieldTypeHandleThrowing();
+    MethodTable* pFieldMT = typeHandle.GetMethodTable();
+    if (typeHandle.IsCanonicalSubtype() || pFieldMT->IsSharedByGenericInstantiations() || pFieldMT->ContainsPointers())
+    {
+        return NULL;
+    }
+
+    PTR_BYTE basePtr = field->GetBase();
+    if (basePtr == nullptr)
+    {
+        return NULL;
+    }
```

Comment:
```suggestion
    _ASSERTE(basePtr != nullptr);
```
I do not think this can be ever NULL.

Reply:
hm.. [apparently](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-77737-merge-8b0b8a25feb1442fbd/System.Text.RegularExpressions.Tests/1/console.1988e590.log?helixlogtype=result) it can? 
```
Assert failure(PID 6901 [0x00001af5], Thread: 55970 [0xdaa2]): basePtr != nullptr
```

Reply:
Perhaps, in case of `pMT->IsDynamicStatics()` ?

Reply:
`IsDynamicStatics` should be eligible for this optimization. I think it is probably something else.